### PR TITLE
[Data Grid] Row validation

### PR DESCRIPTION
**Pull request by ZarkoPernar**
### TODO
- changelog

### Description
Continuation of the work started in #1038. Although the the cell validation api introduced in that PR allows for row-level validation, it was too cumbersome for developers to use for row validation. This PR adds a new api on the Grid component, `getRowValidationStatus` that enables similar styling to cell validation but on a row level, and introduces a new component, `RowValidationStatusColumn` that displays a StatusIcon based on the row validation context value.

### Changes
- new grid context `RowValidationStatusContext`
- new component `RowValidationStatusColumn`
- new `Grid` prop ```getRowValidationStatus?: (
    row: GridRowModel<T>
  ) => CellValidationState | undefined;```
- new `status` characteristics necessary for selected status rows

### Screenshots
localhost_6006_iframe html_args= id=data-grid-data-grid--cell-and-row-validation viewMode=story
localhost_6006_iframe html_args= id=data-grid-data-grid--row-validation viewMode=story (1)

### CODE REVIEW

File: packages/theme/css/characteristics/status.css [hunk 1]
```diff
@@ -13,4 +13,12 @@
...
   --salt-status-warning-background-emphasize: var(--salt-palette-warning-background-emphasize);
   --salt-status-error-background-emphasize: var(--salt-palette-error-background-emphasize);
+
+  --salt-status-error-background-selected: rgba(255, 187, 178, 1);
+  --salt-status-warning-background-selected: rgba(254, 214, 142, 1);
+  --salt-status-success-background-selected: rgba(160, 221, 164, 1);
+
+  --salt-status-error-border-selected: rgba(227, 43, 22, 1);
+  --salt-status-warning-border-selected: rgba(214, 85, 19, 1);
+  --salt-status-success-border-selected: rgba(36, 135, 75, 1);
```
Comment (libertymayc) [inline packages/theme/css/characteristics/status.css:23]:
```suggestion
```

File: packages/grid/src/internal/Cell.css [hunk 2]
```diff
@@ -1,6 +1,6 @@
 .saltGridCell {
   box-sizing: border-box;
-  overflow: hidden;
+  /* overflow: hidden; */
```
[the remark this answers is not in the document]
Comment (ZarkoPernar) [reply packages/grid/src/internal/Cell.css:3]: Yep, good catch

**Comment by codified-likeness-utility** [comment]
Have you tested what the performance is like on a data set of 1000+ rows?  We have a similar implementation on our team that uses AG-Grid and the performance suffers exponentially around 1-2k rows or o (depending on the number of columns). I am curious what the impact will be on this Grid (if any).

**Comment by ZarkoPernar** [comment]
> Have you tested what the performance is like on a data set of 1000+ rows?  We have a similar implementation on our team that uses AG-Grid and the performance suffers exponentially around 1-2k rows or o (depending on the number of columns). I am curious what the impact will be on this Grid (if any).

@codified-likeness-utility validation itself should not impact performance at all since it is only run on the small subset of rows that are being rendered by the virtualization. The validation getters are not called on non-rendered rows.

**Pull request merged**